src/ucp/core/ucp_rkey.c

```diff
@@ -791,6 +850,11 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rkey_proto_resolve,
     rkey_config_key.mem_type           = rkey->mem_type;
     rkey_config_key.unreachable_md_map = unreachable_md_map;
 
+    if (!legacy && (buffer >= buffer_end) &&
+        (ucp_ep_config(ep)->key.dst_version > 19)) {
+        buffer_end = (void*)UINTPTR_MAX;
+    }
```

Comment:
1. can we unite it with the condition in line 858?
2. why need "legacy" flag? can't just check peer version?